serialize/src/lib.rs

```diff
@@ -313,6 +313,75 @@ impl<T: CanonicalSerialize> CanonicalSerialize for [T] {
     }
 }
 
+impl<T: CanonicalSerialize, const N: usize> CanonicalSerialize for [T; N] {
+    #[inline]
+    fn serialize<W: Write>(&self, mut writer: W) -> Result<(), SerializationError> {
+        for item in self.iter() {
+            item.serialize(&mut writer)?;
+        }
+        Ok(())
+    }
+
+    #[inline]
+    fn serialized_size(&self) -> usize {
+        self.iter()
+            .map(|item| item.serialized_size())
+            .sum::<usize>()
+    }
+
+    #[inline]
+    fn serialize_uncompressed<W: Write>(&self, mut writer: W) -> Result<(), SerializationError> {
+        for item in self.iter() {
+            item.serialize_uncompressed(&mut writer)?;
+        }
+        Ok(())
+    }
+
+    #[inline]
+    fn serialize_unchecked<W: Write>(&self, mut writer: W) -> Result<(), SerializationError> {
+        for item in self.iter() {
+            item.serialize_unchecked(&mut writer)?;
+        }
+        Ok(())
+    }
+
+    #[inline]
+    fn uncompressed_size(&self) -> usize {
+        self.iter()
+            .map(|item| item.uncompressed_size())
+            .sum::<usize>()
+    }
+}
+
+impl<T: CanonicalDeserialize + Default + Copy, const N: usize> CanonicalDeserialize for [T; N] {
```

Comment:
I believe we can use new array methods to eliminate the `Default + Copy` bounds:
```rust
Ok([(); N].map(|_| T::deserialize(&mut reader).unwrap()))
```

There is a downside that this panics immediately if there's an error, but that's not necessarily a bad thing. This might be the best we can do until [this feature](https://github.com/rust-lang/rust/issues/89379) is stabilized.

Reply:
Awesome! I changed de-serialization to do this, removed the extra bounds, and added a `TODO` to update things once the feature is stabilized.